tests/integration/test_schema_compatibility.py

```diff
@@ -0,0 +1,219 @@
+"""
+Copyright (c) 2024 Aiven Ltd
+See LICENSE for details
+"""
+from collections.abc import Coroutine
+from dataclasses import dataclass
+from karapace.client import Client
+from karapace.typing import Subject
+from tests.base_testcase import BaseTestCase
+from typing import Any, Callable, Union
+
+import json
+import logging
+import pytest
+
+LOG = logging.getLogger(__name__)
+
+schema_int = {"type": "record", "name": "schema_name", "fields": [{"type": "int", "name": "field_name"}]}
+schema_long = {"type": "record", "name": "schema_name", "fields": [{"type": "long", "name": "field_name"}]}
+schema_string = {"type": "record", "name": "schema_name", "fields": [{"type": "string", "name": "field_name"}]}
+schema_double = {"type": "record", "name": "schema_name", "fields": [{"type": "double", "name": "field_name"}]}
+
+
+@dataclass
+class SchemaCompatibilityTestCase(BaseTestCase):
+    new_schema: str
+    compatibility_mode: str
+    register_baseline_schemas: Callable[[Client, Subject], Coroutine[Any, Any, None]]
+    expected_is_compatible: Union[bool, None]
+    expected_status_code: int
+    expected_incompatibilities: Union[str, None]
```

Comment:
same here, but should be fixed by the `pre-commit` as soon as you import the `annotations`